### PR TITLE
Add 0x prefix to Rosetta Account Alias

### LINKED ISSUE #3855: Rosetta add 0x prefix to alias addresses
### Problem

It’s a common practice to prefix a hex string with 0x. The Rosetta AddressIdentifier.Address when using an account’s alias ed25519 public key doesn’t have the prefix, we need to add it

### Solution

Add the 0x prefix to the hex encoded account alias ed25519 public key string.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by edwin-greene**
**Description**:
Add the 0x prefix to the Account Alias ed25519 public key string.

**Related issue(s)**:

Fixes #3855

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rosetta/app/domain/types/account_id.go [hunk 1]
```diff
@@ -73,7 +74,7 @@ func (a AccountId) IsZero() bool {
 
 func (a AccountId) String() string {
 	if a.HasAlias() {
-		return hex.EncodeToString(a.alias)
+		return tools.SafeAddHexPrefix(hex.EncodeToString(a.alias))
```
[the remark this answers is not in the document]
Comment (edwin-greene) [reply hedera-mirror-rosetta/app/domain/types/account_id.go:77]: I've added a change that will remove it.

File: hedera-mirror-rosetta/app/domain/types/account_id.go [hunk 2]
```diff
@@ -101,7 +102,7 @@ func NewAccountIdFromString(address string, shard, realm int64) (AccountId, erro
 		}
 		return AccountId{accountId: entityId}, nil
 	}
-	alias, err := hex.DecodeString(address)
+	alias, err := hex.DecodeString(tools.SafeRemoveHexPrefix(address))
```
Comment (xin-hedera) [inline hedera-mirror-rosetta/app/domain/types/account_id.go:105]: I meant we should enforce the alias string passed in has the `0x` prefix. If there's no prefix we should return error.

**Review by xin-hedera — CHANGES_REQUESTED** [review]

**Review by xin-hedera — APPROVED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Pull request merged**